src/bin/common.rs

```diff
@@ -527,6 +532,24 @@ fn parse_config(matches: &CliOptions) -> Result<EncoderConfig, CliError> {
 
   let mut cfg = EncoderConfig::with_speed_preset(speed);
 
+  if let Some(level_str) = &matches.level {
+    cfg.level_idx = match &level_str.as_str() {
+      &"auto" => None,
+      &"unconstrained" => Some(31),
+      _ => {
+        let (major, minor) = scan_fmt!(&level_str, "{}.{}", u8, u8)
+          .expect("Could not parse AV1 level");
+        if major > 7 {
+          panic!("Invalid major AV1 level")
```

Comment:
Just my personal preference but I probably wouldn't bother specifying major or minor here, it's a bit arbitrary.